### ISSUE TITLE
Adopt client to latest 5.0+ sync-reporting API

### ISSUE THREAD

**Issue by dunossauro** [reportportal/client-Python#53]
I'm trying running cline-python current version on docker-compose and has many errors related a request.

for example:
```
{'errorCode': 4001, 'message': "Incorrect Request. [Field 'startTime' should not be null.] "}
```
Or
```
{'errorCode': 4001, 'message': "Incorrect Request. [Field 'stopTime' should not be null.] "}
```

When investigating that I found in `ReportPortalService` many calls with the wrong payload. Like that:

```Python
    def start_launch(self, name, start_time, description=None, tags=None,
                     mode=None):
        data = {
            "name": name,
            "description": description,
            "tags": tags,
            "start_time": start_time,
            "mode": mode
        }
```

Maybe the API change, but I don't know exactly, because yesterday I started my first tests with report portal.

**Comment by dunossauro** [comment]
Trying fix that: https://github.com/reportportal/client-Python/compare/master...dunossauro:master.

It's wrong implemantation, but works for me. I dont test log yet.

**Comment by dunossauro** [comment]
After work all day in fix requests I discover the problem. The reportportal.io/download get the beta version. Works good on version 4.

**Comment by DzmitryHumianiuk** [comment]
@dunossauro that's right.
current client made for RPv4.
RPv5 has significant difference with v4 reporting API.

BUT
**we added `full backward compatibility` support for v4 API in RPv5.**
so you can give RPv5 a try with client-python v4.

please let us know your results.

latest deployed v5 instance can be found at https://beta.demo.reportportal.io

**Comment by dunossauro** [comment]
@DzmitryHumianiuk, Thanks for answering me.

I'll try run in v5 and I say in this issue my resultas. Thanks for your time

**Comment by qkiss** [comment]
Has this issue been addressed? I'm trying to find the client-python v4 version but don't see it.

**Comment by DzmitryHumianiuk** [comment]
version [3.2.3](https://pypi.org/project/reportportal-client/3.2.3/) is for v4 of RP.

v5 of ReportPortal should also accept v4-based results

**Comment by qkiss** [comment]
I am trying to use RobotFramework agent and it only works with couple of patches from [master...dunossauro:master](https://github.com/reportportal/client-Python/compare/master...dunossauro:master). For me it's fine for now.

Also I don't see where it's stated that:
**we added full backward compatibility support for v4 API in RPv5.**

**Comment by DzmitryHumianiuk** [comment]
@qkiss what's your version of ReportPortal you use right now?

>we added full backward compatibility support for v4 API in RPv5.
that means, that if your test integration worked with latest v4, that should work with latest v5 as well.

**Comment by qkiss** [comment]
I have 5.0.0-BETA-12. Will try with the latest one.

**Comment by DzmitryHumianiuk** [comment]
@qkiss beta-12 should also work with original v4 based reporting.
without that fix

**Comment by iivanou** [comment]
Closed as already fixed in the client version 5.0.0.
